lib/kino_slack/message_interpolator.ex

```diff
@@ -0,0 +1,60 @@
+defmodule KinoSlack.MessageInterpolator do
+  def interpolate(message) do
+    ast = quote do: <<"">>
+    interpolate(message, ast)
+  end
+
+  defp interpolate("", result_ast) do
+    result_ast
+  end
+
+  defp interpolate("{{" <> rest, ast) do
+    with [inner, rest] <- String.split(rest, "}}", parts: 2),
+         {:ok, expression} <- Code.string_to_quoted(inner) do
+      ast = append_interpolation(ast, expression)
+      interpolate(rest, ast)
+    else
+      _ ->
+        <<char::utf8>> = "{"
+        ast = append_char(ast, char)
+        ast = append_char(ast, char)
+        interpolate(rest, ast)
+    end
+  end
+
+  defp interpolate(<<char::utf8, rest::binary>>, ast) do
+    new_ast = append_char(ast, char)
+    interpolate(rest, new_ast)
+  end
+
+  defp append_interpolation(ast, expression) do
+    interpolation_node = {
+      :"::",
+      [],
+      [
+        {{:., [], [Kernel, :to_string]}, [], [expression]},
+        {:binary, [], Elixir}
+      ]
+    }
+
+    {_, _, args} = ast
+    args = args ++ [interpolation_node]
+
+    {:<<>>, [], args}
+  end
+
+  defp append_char(ast, char) do
```

Comment:
It may simplify a bit if you make it `append_string`.

Reply:
Nice tip. ✅